File: h1.cc
#include <iostream>

using namespace std;

int main(int cantidad, char* argumentos[]) {
    if (cantidad < 2) {
        cout << "Error: No hay argumentos" << endl;
    }
    for ( int i=cantidad-1; i > 0; --i) {
        cout << argumentos[i] << endl;
    }

}

File: H1.java
public class H1 {
    public static void main(String[] argumentos) {
        if (argumentos.length < 1) {
           System.out.println("Error: No hay argumentos");
            
        }

        int i = argumentos.length - 1;
        for (; i >= 0; i--) {
            System.out.println(argumentos[i]);
        }
    }
}